SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -260,12 +260,20 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 
 Worker_ComponentData USpatialSender::CreateLevelComponentData(AActor* Actor)
 {
-	if (FLevelData* LevelData = ClassInfoManager->SchemaDatabase->LevelPathToLevelData.Find(NetDriver->World->GetName()))
+	UWorld* ActorWorld = Actor->GetTypedOuter<UWorld>();
+	if (ActorWorld != NetDriver->World)
 	{
-		if (uint32* ComponentId = LevelData->SublevelNameToComponentId.Find(Actor->GetTypedOuter<UWorld>()->GetName()))
+		FString CleanLevelPath = UWorld::RemovePIEPrefix(ActorWorld->GetOuter()->GetPathName());
+		uint32* ComponentId = ClassInfoManager->SchemaDatabase->LevelPathToComponentId.Find(CleanLevelPath);
```

Comment:
More variable declaration into if statement

Reply:
Could also add a GetComponentIdFromLevelPath() func to SchemaDatabase that does the clean for us, as it's not valid to call it with a dirty path.